s3/s3_plugin_logging_category.hpp

```diff
@@ -0,0 +1,37 @@
+#ifndef S3_PLUGIN_LOGGING_CATEGORY
+#define S3_PLUGIN_LOGGING_CATEGORY
+
+
+#include "irods/irods_logger.hpp"
+
+// 1. Declare the custom category tag.
+//    This structure does not need to define a body.
+//    This tag allows the logger to locate data specific to the new category.
+struct s3_plugin_logging_category;
+
+// 2. Specialize the logger configuration for the new category.
+//    This also defines the default configuration for the new category.
+namespace irods::experimental
+{
+    template <>
+    class log::logger_config<s3_plugin_logging_category>
+    {
+        // This defines the name that will appear in the log under the "log_category" key.
+        // The "log_category" key defines where the message originated. Try to use a name
+        // that makes it easy for administrators to determine what produced the message.
+        static constexpr const char* name = "s3_plugin_logging_category";
+
+        // This is required since the fields above are private.
+        // This allows the logger to access and modify the configuration.
+        friend class logger<s3_plugin_logging_category>;
+
+    public:
+        // This is the current log level for the category. This also represents the initial
+        // log level. Use the "set_level()" function to adjust the level.
+        static inline log::level level = log::level::info;
+
+    };
```

Comment:
@korydraughn, we have some convenience types declared in `irods_logger.hpp` which might be useful here. Do you think it would be appropriate to do so for these new categories somewhere in the header file?

https://github.com/irods/irods/blob/eb361f5e49e817ff498dbb417eb935030e5b2322/lib/core/include/irods/irods_logger.hpp#L74-L85

Reply:
Are you saying to define them in `irods_logger.hpp` or this particular header file?

I think it is perfectly fine to define type aliases for the s3 log category in this file. It would make it a little more convenient to use the logger then.

The type alias should be defined outside of the experimental namespace unless it's truly experimental.

Reply:
Right, I feel like this is more in line with what I see in irods/irods:
```c++
using s3_logger = irods::experimental::log::s3_plugin;
```
Which would act as a replacement for:
```c++
using s3_logger = irods::experimental::log::logger<s3_plugin_logging_category>;
```

Reply:
We can argue about the naming, just wanted to throw out that the type alias would be nice to introduce here.

Reply:
Talked to Alan.  We determined that this might not be possible due to how this is set up.